### PR TITLE
[GU-API] HUB info

### PR DESCRIPTION
**Pull request by prekucki**

### CODE REVIEW

File: gu-hub-api.yaml [hunk 1]
```diff
@@ -454,6 +467,43 @@ paths:
...
+        example: '0x875f272d3b9e7b55a5784a131a60bf3d7a42c73c'
+        readOnly: true
+      version:
+        type: string
+        description: semver of hub.
+        example: '0.2.1'
+      build:
+        $ref: '#/definitions/BuildInfo'
+      caps:
+        description: hub protocool capabilities
```
Comment (quilir) [inline gu-hub-api.yaml]:
```suggestion
        description: hub protocol capabilities
```

File: gu-hub-api.yaml [hunk 2]
```diff
@@ -454,6 +467,43 @@ paths:
...
+        type: string
+        description: git commit id
+
+  HubInfo:
+    description: General information about given HUB.
+    type: object
+    properties:
+      nodeId:
+        type: string
+        description: node public key hash in etherium format
```
Comment (quilir) [inline gu-hub-api.yaml]:
```suggestion
        description: node public key hash in ethereum format
```

File: gu-hub/src/local_service.rs [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+            }
+            actix_web::Either::A(HttpResponse::NotFound().body("err"))
+        })
+    }
+}
+
+struct ProxyHandler(Arc<RwLock<BTreeMap<String, BTreeMap<String, String>>>>);
+
+#[derive(Debug)]
+struct ServiceRunner {}
```
Comment (quilir) [inline gu-hub/src/local_service.rs:211]: Never used struct

File: gu-hub/webapp/directives/guListProvider.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
+angular.module('gu').directive('guListProvider', function () {
+    return {
+        restrict: 'E',
+        scope: {
+            peers: '=ngModel'
+        },
+        require: 'ngModel',
+        templateUrl: 'directives/guListProvider.html',
+        controller: function ($scope, sessionMan, $q) {
+            console.log('ddd');
```
Comment (quilir) [inline gu-hub/webapp/directives/guListProvider.js]: Left debug log

File: gu-plugin-api/src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+            client
+                .request(Method::PATCH, uri)
+                .send_json(&command)
+                .into_future()
+                .map_err(|e| eprintln!("hub connection error: {}", e))
+        })
+        .and_then(|mut r| {
+            JsonBody::new(&mut r).map_err(|e| eprintln!("hub connection error: {}", e))
+        })
+        .and_then(|v: serde_json::Value| Ok(log::info!("registed service [{}]", v.to_string())))
```
Comment (quilir) [inline gu-plugin-api/src/lib.rs]:
```suggestion
        .and_then(|v: serde_json::Value| Ok(log::info!("registered service [{}]", v.to_string())))
```

File: gu-plugin-api/src/lib.rs [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+            .scheme("http")
+            .authority(addr.parse::<http::uri::Authority>()?)
+            .path_and_query("/service/local")
+            .build(),
+        Err(_addr) => Ok(Uri::from_static("http://127.0.0.1:61622/service/local")),
+    }
+}
+
+/// Registers command service in hub registry.
+pub fn register_server(url: &str, cmd_name: &str) -> impl Future<Item = (), Error = ()> {
```
Comment (quilir) [inline gu-plugin-api/src/lib.rs]:
```suggestion
pub fn register_service(url: &str, cmd_name: &str) -> impl Future<Item = (), Error = ()> {
```

**Review by quilir — CHANGES_REQUESTED** [review]

File: gu-hub/src/local_service.rs (hunk 3 above)
Comment (prekucki) [reply gu-hub/src/local_service.rs:211]: I know. Implementation pending. Service runner should activate local services relevant to active plugins.

**Pull request merged**